app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @users = [User.find(params[:user_id])]
+    @users = [User.includes(:posts).find(params[:user_id])]
     @posts = @users[0].posts
```

Comment:
- You are only fetching the User's posts on this line but not post comments as well. Please update this to:
`@posts = @users[0].posts.includes(:comments)`